File: tests/examples/helloworld/helloworld.c
#include <gst/gst.h>

static gboolean
bus_call (GstBus * bus, GstMessage * msg, gpointer data)
{
  GMainLoop *loop = (GMainLoop *) data;

  switch (GST_MESSAGE_TYPE (msg)) {
    case GST_MESSAGE_EOS:{
      g_print ("End-of-stream\n");
      g_main_loop_quit (loop);
      break;
    }
    case GST_MESSAGE_ERROR:{
      gchar *debug;
      GError *err;

      gst_message_parse_error (msg, &err, &debug);
      g_free (debug);

      g_print ("Error: %s\n", err->message);
      g_error_free (err);

      g_main_loop_quit (loop);

      break;
    }
    default:
      break;
  }
  return TRUE;
}

gint
main (gint argc, gchar * argv[])
{
  GstElement *playbin;
  GMainLoop *loop;
  GstBus *bus;
  gchar *uri;

  gst_init (&argc, &argv);

  if (argc < 2) {
    g_print ("usage: %s <media file or uri>\n", argv[0]);
    return 1;
  }

  playbin = gst_element_factory_make ("playbin2", NULL);
  if (!playbin) {
    g_print ("'playbin2' gstreamer plugin missing\n");
    return 1;
  }

  /* take the commandline argument and ensure that it is a uri */
  if (gst_uri_is_valid (argv[1]))
    uri = g_strdup (argv[1]);
  else
    uri = gst_filename_to_uri (argv[1], NULL);
  g_object_set (playbin, "uri", uri, NULL);
  g_free (uri);

  /* create and event loop and feed gstreamer bus mesages to it */
  loop = g_main_loop_new (NULL, FALSE);

  bus = gst_element_get_bus (playbin);
  gst_bus_add_watch (bus, bus_call, loop);
  g_object_unref (bus);

  /* start play back and listed to events */
  gst_element_set_state (playbin, GST_STATE_PLAYING);
  g_main_loop_run (loop);

  /* cleanup */
  gst_element_set_state (playbin, GST_STATE_NULL);
  g_object_unref (playbin);
  g_main_loop_unref (loop);

  return 0;
}

File: old_examples/helloworld.py
#!/usr/bin/env python

import sys

import gobject
gobject.threads_init()

import pygst
pygst.require('0.10')
import gst

def bus_call(bus, message, loop):
    t = message.type
    if t == gst.MESSAGE_EOS:
        sys.stout.write("End-of-stream\n")
        loop.quit()
    elif t == gst.MESSAGE_ERROR:
        err, debug = message.parse_error()
        sys.stderr.write("Error: %s: %s\n" % err, debug)
        loop.quit()
    return True

def main(args):
    if len(args) != 2:
        sys.stderr.write("usage: %s <media file or uri>\n" % args[0])
        sys.exit(1)
        
    playbin = gst.element_factory_make("playbin2", None)
    if not playbin:
        sys.stderr.write("'playbin2' gstreamer plugin missing\n")
        sys.exit(1)

    # take the commandline argument and ensure that it is a uri
    if gst.uri_is_valid(args[1]):
      uri = args[1]
    else:
      uri = gst.filename_to_uri(args[1])
    playbin.set_property('uri', uri)

    # create and event loop and feed gstreamer bus mesages to it
    loop = gobject.MainLoop()

    bus = playbin.get_bus()
    bus.add_watch(bus_call, loop)
    
    # start play back and listed to events
    playbin.set_state(gst.STATE_PLAYING)
    loop.run()
    
    # cleanup
    playbin.set_state(gst.STATE_NULL)

if __name__ == '__main__':
    sys.exit(main(sys.argv))
